src/praisonai/praisonai/integrations/managed_agents.py

```
@@ -0,0 +1,425 @@
+        if session_id is None:
+            agent_id = await self._ensure_agent()
+            environment_id = await self._ensure_environment()
+            session_id = await self.backend.create_session(agent_id, environment_id)
+            self._session_cache[session_key] = session_id
+        
+        # Send message
+        await self.backend.send_message(session_id, prompt)
+        
+        # Stream events with timeout
+        try:
+            async for event in asyncio.wait_for(
```

Ruff (0.15.9)

[error] Undefined name `asyncio`

(F821)